src/mono/mono/mini/aot-compiler.c

```diff
@@ -13718,6 +13774,33 @@ got_info_free (GotInfo *info)
 	g_ptr_array_free (info->got_patches, TRUE);
 }
 
+static void
+aot_opts_free (MonoAotOptions *aot_opts)
+{
+	g_free (aot_opts->outfile);
+	g_free (aot_opts->llvm_outfile);
+	g_free (aot_opts->data_outfile);
+	g_list_free (aot_opts->profile_files);
+	g_list_free (aot_opts->mibc_profile_files);
+	g_free (aot_opts->gen_msym_dir_path);
+	g_list_free (aot_opts->direct_pinvokes);
```

Comment:
This will just free the list and its nodes, but not the data, if data on each nodes needs to be free as well you can implement something like this:

```
for (GList *it = list; it; it = it->next)
    g_free (it->data);
g_list_free (list);
```

Reply:
Thanks, added the for loop to free node data for each GList in MonoAotOptions